### PR TITLE
OEL-603: Create OpenEuropa Event

### PR DESCRIPTION
**Pull request by GilNovacomm**
### Description

Initial development of the OE Showcase event.

### Change log

- Added:
- Changed:
- Deprecated:
- Removed:
- Fixed:
- Security:

### Commands

```sh
[Insert commands here]

```

### CODE REVIEW

File: modules/oe_showcase_event/oe_showcase_event.info.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  - rdf_skos:rdf_skos
+  - address:address
+  - link_attributes:link_attributes
+  - drupal:datetime_range
+  - drupal:file
+  - drupal:image
+  - facets_form_date_range:facets_form_date_range
+  - search_api:search_api
+  - search_api_db:search_api_db
+
```
Comment (escuriola) [inline modules/oe_showcase_event/oe_showcase_event.info.yml:18]: add ckeditor dependency

File: modules/oe_showcase_event/README.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
+# OpenEuropa Showcase Page
```
Comment (abel-santos-corral) [inline modules/oe_showcase_event/README.md]: Rename this.

File: modules/oe_showcase_event/oe_showcase_event.install [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+ * @file
+ * The OpenEuropa Contact Forms installation file.
+ */
+
+declare(strict_types = 1);
+
+/**
+ * Implements hook_install().
+ */
+function oe_showcase_event_install($is_syncing) {
```
Comment (abel-santos-corral) [inline modules/oe_showcase_event/oe_showcase_event.install]: Add the ": void" in the function.

File: modules/oe_showcase_event/config/install/field.storage.node.field_oe_showcase_event_url.yml [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    - node
+id: node.field_oe_showcase_event_url
+field_name: field_oe_showcase_event_url
+entity_type: node
+type: link
+settings: {  }
+module: link
+locked: false
+cardinality: 1
+translatable: true
```
Comment (abel-santos-corral) [inline modules/oe_showcase_event/config/install/field.storage.node.field_oe_showcase_event_url.yml:15]:
```suggestion
translatable: false
```

File: modules/oe_showcase_event/config/install/field.storage.node.field_oe_showcase_event_registra.yml [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+    - node
+id: node.field_oe_showcase_event_registra
+field_name: field_oe_showcase_event_registra
+entity_type: node
+type: link
+settings: {  }
+module: link
+locked: false
+cardinality: 1
+translatable: true
```
Comment (abel-santos-corral) [inline modules/oe_showcase_event/config/install/field.storage.node.field_oe_showcase_event_registra.yml:15]:
```suggestion
translatable: false
```

File: modules/oe_showcase_event/config/install/field.storage.node.field_oe_showcase_event_document.yml [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+type: file
+settings:
+  display_field: false
+  display_default: false
+  uri_scheme: public
+  target_type: file
+module: file
+locked: false
+cardinality: -1
+translatable: true
```
Comment (abel-santos-corral) [inline modules/oe_showcase_event/config/install/field.storage.node.field_oe_showcase_event_document.yml:19]:
```suggestion
translatable: false
```

File: modules/oe_showcase_event/config/install/field.storage.node.field_oe_showcase_event_date.yml [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+id: node.field_oe_showcase_event_date
+field_name: field_oe_showcase_event_date
+entity_type: node
+type: daterange
+settings:
+  datetime_type: datetime
+module: datetime_range
+locked: false
+cardinality: 1
+translatable: true
```
Comment (abel-santos-corral) [inline modules/oe_showcase_event/config/install/field.storage.node.field_oe_showcase_event_date.yml:16]:
```suggestion
translatable: false
```

File: modules/oe_showcase_event/config/install/field.field.node.oe_showcase_event.field_oe_showcase_event_url.yml [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+  module:
+    - link
+id: node.oe_showcase_event.field_oe_showcase_event_url
+field_name: field_oe_showcase_event_url
+entity_type: node
+bundle: oe_showcase_event
+label: 'Event URL'
+description: ''
+required: false
+translatable: true
```
Comment (abel-santos-corral) [inline modules/oe_showcase_event/config/install/field.field.node.oe_showcase_event.field_oe_showcase_event_url.yml]:
```suggestion
translatable: false
```

File: modules/oe_showcase_event/config/install/field.field.node.oe_showcase_event.field_oe_showcase_event_location.yml [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+  module:
+    - address
+id: node.oe_showcase_event.field_oe_showcase_event_location
+field_name: field_oe_showcase_event_location
+entity_type: node
+bundle: oe_showcase_event
+label: 'Event Location'
+description: ''
+required: false
+translatable: true
```
Comment (abel-santos-corral) [inline modules/oe_showcase_event/config/install/field.field.node.oe_showcase_event.field_oe_showcase_event_location.yml:16]:
```suggestion
translatable: false
```

File: modules/oe_showcase_event/config/install/field.field.node.oe_showcase_event.field_oe_showcase_event_image.yml [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+  module:
+    - image
+id: node.oe_showcase_event.field_oe_showcase_event_image
+field_name: field_oe_showcase_event_image
+entity_type: node
+bundle: oe_showcase_event
+label: Image
+description: ''
+required: false
+translatable: true
```
Comment (abel-santos-corral) [inline modules/oe_showcase_event/config/install/field.field.node.oe_showcase_event.field_oe_showcase_event_image.yml]:
```suggestion
translatable: false
```

File: modules/oe_showcase_event/config/install/field.field.node.oe_showcase_event.field_oe_showcase_event_document.yml [hunk 11]
```diff
@@ -0,0 +1,26 @@
...
+  module:
+    - file
+id: node.oe_showcase_event.field_oe_showcase_event_document
+field_name: field_oe_showcase_event_document
+entity_type: node
+bundle: oe_showcase_event
+label: Documents
+description: ''
+required: false
+translatable: true
```
Comment (abel-santos-corral) [inline modules/oe_showcase_event/config/install/field.field.node.oe_showcase_event.field_oe_showcase_event_document.yml]:
```suggestion
translatable: false
```

File: modules/oe_showcase_event/config/install/field.field.node.oe_showcase_event.field_oe_showcase_event_date.yml [hunk 12]
```diff
@@ -0,0 +1,20 @@
...
+  module:
+    - datetime_range
+id: node.oe_showcase_event.field_oe_showcase_event_date
+field_name: field_oe_showcase_event_date
+entity_type: node
+bundle: oe_showcase_event
+label: 'Event Date'
+description: ''
+required: false
+translatable: true
```
Comment (abel-santos-corral) [inline modules/oe_showcase_event/config/install/field.field.node.oe_showcase_event.field_oe_showcase_event_date.yml]:
```suggestion
translatable: false
```

File: modules/oe_showcase_event/config/install/core.entity_view_display.node.oe_showcase_event.default.yml [hunk 13]
```diff
@@ -0,0 +1,108 @@
...
+    - text
+    - user
+id: node.oe_showcase_event.default
+targetEntityType: node
+bundle: oe_showcase_event
+mode: default
+content:
+  field_oe_showcase_event_content:
+    weight: 101
+    label: above
```
Comment (abel-santos-corral) [inline modules/oe_showcase_event/config/install/core.entity_view_display.node.oe_showcase_event.default.yml]:
```suggestion
    label: hidden
```

File: modules/oe_showcase_event/config/install/core.entity_view_display.node.oe_showcase_event.default.yml [hunk 14]
```diff
@@ -0,0 +1,108 @@
...
+    weight: 104
+    label: above
+    settings:
+      use_description_as_link_text: true
+    third_party_settings: {  }
+    type: file_default
+    region: content
+  field_oe_showcase_event_image:
+    weight: 105
+    label: above
```
Comment (abel-santos-corral) [inline modules/oe_showcase_event/config/install/core.entity_view_display.node.oe_showcase_event.default.yml]:
```suggestion
    label: hidden
```

File: modules/oe_showcase_event/config/install/core.entity_view_display.node.oe_showcase_event.search_result.yml [hunk 15]
```diff
@@ -0,0 +1,113 @@
...
+    - text
+    - user
+id: node.oe_showcase_event.search_result
+targetEntityType: node
+bundle: oe_showcase_event
+mode: search_result
+content:
+  field_oe_showcase_event_content:
+    weight: 101
+    label: above
```
Comment (abel-santos-corral) [inline modules/oe_showcase_event/config/install/core.entity_view_display.node.oe_showcase_event.search_result.yml]:
```suggestion
    label: hidden
```

File: modules/oe_showcase_event/config/install/core.entity_view_display.node.oe_showcase_event.search_result.yml [hunk 16]
```diff
@@ -0,0 +1,113 @@
...
+    weight: 104
+    label: above
+    settings:
+      use_description_as_link_text: true
+    third_party_settings: {  }
+    type: file_default
+    region: content
+  field_oe_showcase_event_image:
+    weight: 105
+    label: above
```
Comment (abel-santos-corral) [inline modules/oe_showcase_event/config/install/core.entity_view_display.node.oe_showcase_event.search_result.yml]:
```suggestion
    label: hidden
```

File: modules/oe_showcase_event/config/install/core.entity_view_display.node.oe_showcase_event.teaser.yml [hunk 17]
```diff
@@ -0,0 +1,109 @@
...
+    - text
+    - user
+id: node.oe_showcase_event.teaser
+targetEntityType: node
+bundle: oe_showcase_event
+mode: teaser
+content:
+  field_oe_showcase_event_content:
+    weight: 101
+    label: above
```
Comment (abel-santos-corral) [inline modules/oe_showcase_event/config/install/core.entity_view_display.node.oe_showcase_event.teaser.yml]:
```suggestion
    label: hidden
```

File: modules/oe_showcase_event/config/install/core.entity_view_display.node.oe_showcase_event.teaser.yml [hunk 18]
```diff
@@ -0,0 +1,109 @@
...
+    weight: 104
+    label: above
+    settings:
+      use_description_as_link_text: true
+    third_party_settings: {  }
+    type: file_default
+    region: content
+  field_oe_showcase_event_image:
+    weight: 105
+    label: above
```
Comment (abel-santos-corral) [inline modules/oe_showcase_event/config/install/core.entity_view_display.node.oe_showcase_event.teaser.yml]:
```suggestion
    label: hidden
```

File: modules/oe_showcase_event/config/install/core.entity_view_display.node.oe_showcase_event.search_result.yml [hunk 19]
```diff
@@ -0,0 +1,113 @@
...
+    settings: {  }
+    third_party_settings: {  }
+    type: text_default
+    region: content
+  field_oe_showcase_event_date:
+    weight: 102
+    label: above
+    settings:
+      separator: '-'
+      format_type: medium
```
Comment (abel-santos-corral) [inline modules/oe_showcase_event/config/install/core.entity_view_display.node.oe_showcase_event.search_result.yml]:
```suggestion
      format_type: html_date
```

File: modules/oe_showcase_event/config/install/core.entity_view_display.node.oe_showcase_event.default.yml [hunk 20]
```diff
@@ -0,0 +1,108 @@
...
+    settings: {  }
+    third_party_settings: {  }
+    type: text_default
+    region: content
+  field_oe_showcase_event_date:
+    weight: 102
+    label: above
+    settings:
+      separator: '-'
+      format_type: medium
```
Comment (abel-santos-corral) [inline modules/oe_showcase_event/config/install/core.entity_view_display.node.oe_showcase_event.default.yml]:
```suggestion
      format_type: short
```

**Review by abel-santos-corral — CHANGES_REQUESTED** [review]
@GilNovacomm please, configure a default image as we did per News content type.

File: modules/oe_showcase_event/config/install/field.field.node.oe_showcase_event.field_oe_showcase_event_registra.yml [hunk 21]
```diff
@@ -0,0 +1,22 @@
...
+  module:
+    - link
+id: node.oe_showcase_event.field_oe_showcase_event_registra
+field_name: field_oe_showcase_event_registra
+entity_type: node
+bundle: oe_showcase_event
+label: 'Event Registration URL'
+description: ''
+required: false
+translatable: false
```
Comment (drishu) [inline modules/oe_showcase_event/config/install/field.field.node.oe_showcase_event.field_oe_showcase_event_registra.yml]: Links can have translations for sure

File: modules/oe_showcase_event/config/install/field.storage.node.field_oe_showcase_event_image.yml [hunk 22]
```diff
@@ -0,0 +1,29 @@
...
+    title: ''
+    width: null
+    height: null
+  target_type: file
+  display_field: false
+  display_default: false
+module: image
+locked: false
+cardinality: 1
+translatable: true
```
Comment (drishu) [inline modules/oe_showcase_event/config/install/field.storage.node.field_oe_showcase_event_image.yml]: false

File: modules/oe_showcase_event/config/install/field.storage.node.field_oe_showcase_event_type.yml [hunk 23]
```diff
@@ -0,0 +1,19 @@
...
+id: node.field_oe_showcase_event_type
+field_name: field_oe_showcase_event_type
+entity_type: node
+type: skos_concept_entity_reference
+settings:
+  target_type: skos_concept
+module: rdf_skos
+locked: false
+cardinality: 1
+translatable: true
```
Comment (drishu) [inline modules/oe_showcase_event/config/install/field.storage.node.field_oe_showcase_event_type.yml]: false

File: modules/oe_showcase_event/oe_showcase_event.info.yml [hunk 24]
```diff
@@ -0,0 +1,43 @@
+name: OpenEuropa Showcase Event
+type: module
+description: 'Demo showcase event feature.'
+core_version_requirement: ^8.9 || ^9.1
+package: OpenEuropa
+
+dependencies:
+  - rdf_skos:rdf_skos
```
Comment (drishu) [inline modules/oe_showcase_event/oe_showcase_event.info.yml]: order alphabetically

File: modules/oe_showcase_event/oe_showcase_event.module [hunk 25]
```diff
@@ -0,0 +1,14 @@
...
+ * @file
+ * Showcase event module.
+ */
+
+/**
+ * Implements link_attributes_plugin_alter().
+ */
+function oe_showcase_event_link_attributes_plugin_alter(&$plugins): void {
+  // Set a default value for the target attribute.
+  $plugins['target']['default_value'] = '_blank';
```
Comment (drishu) [inline modules/oe_showcase_event/oe_showcase_event.module:13]: why do we need this ?

File: modules/oe_showcase_event/config/install/field.field.node.oe_showcase_event.field_oe_showcase_event_image.yml [hunk 26]
```diff
@@ -0,0 +1,37 @@
...
+    - image
+id: node.oe_showcase_event.field_oe_showcase_event_image
+field_name: field_oe_showcase_event_image
+entity_type: node
+bundle: oe_showcase_event
+label: Image
+description: ''
+required: false
+translatable: false
+default_value: {  }
```
Comment (drishu) [inline modules/oe_showcase_event/config/install/field.field.node.oe_showcase_event.field_oe_showcase_event_image.yml:17]: a default image would be good to not set any images for the exported items

**Review by drishu — CHANGES_REQUESTED** [review]

File: modules/oe_showcase_event/oe_showcase_event.module (hunk 25 above)
Comment (GilNovacomm) [reply modules/oe_showcase_event/oe_showcase_event.module:13]: To set _blank as default value, though it is possible to set on the field configuration it is a known bug on the link attributes module.

**Comment by brummbar** [comment]
Closing as outdated.

**Pull request closed without merging**